### PR TITLE
fix: files-from-path with common path prefix removed

### PR DESCRIPTION
**Pull request by alanshaw**
I think this is what we want.

This PR adds a new function `filesFromPaths` that returns a list of File objects with common path prefix removed. i.e. in this example `test` has been removed:

1. Screenshot 2022-12-13 at 15 46 30

2. Screenshot 2022-12-13 at 15 54 32

3. Screenshot 2022-12-13 at 15 53 00

No commonality:

4. Screenshot 2022-12-13 at 15 55 52

**It means that you only ever need `--no-wrap` when there's 1 file**. In the case of multiple files, they are either in different directories and so _must_ be wrapped (1/2), or they are in the same directory, their common prefix is removed and they _must_ be wrapped (3).

It's also consistent with the `FileList` you'd receive from a HTML `<input type="file" multiple/>`.

Added bonus is that this function returns `File` objects whose `stream()` method returns a web `ReadableStream` not a Node.js one, so no need to convert. It also removes globbing dependencies, `util.promisify` and and bunch of other functionality we're not using from `files-from-path`.

### CODE REVIEW

File: lib.js [hunk 1]
```diff
@@ -119,3 +120,89 @@ export async function readProof (path) {
     process.exit(1)
   }
 }
+
+/**
+ * @param {string[]} paths
+ * @param {object} [options]
+ * @param {boolean} [options.hidden]
+ * @returns {Promise<import('@web3-storage/w3up-client/types').FileLike>}
```
Comment (olizilla) [inline lib.js:128]: is return array!
```suggestion
 * @returns {Promise<import('@web3-storage/w3up-client/types').FileLike[]>}
```

File: lib.js [hunk 2]
```diff
@@ -119,3 +120,89 @@ export async function readProof (path) {
...
+ */
+async function * filesFromDir (dir, filter) {
+  const entries = await fs.promises.readdir(path.join(dir), { withFileTypes: true })
+  for (const entry of entries) {
+    if (!filter(entry.name)) {
+      continue
+    }
+
+    if (entry.isFile()) {
+      if (filter(entry.name)) {
```
Comment (olizilla) [inline lib.js:197]: why do we check if the filename is filtered again here when we check it line 192? i think `filter(entry.name)` is sure to be true here?

Comment (alanshaw) [reply lib.js]: fixed!

**Review by olizilla — APPROVED** [review]
its cool i like it

**Comment by olizilla** [comment]
(needs more tests, but can be later)

**Pull request merged**